src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -442,15 +443,16 @@ private static unsafe int ReceiveMessageFrom(SafeCloseSocket socket, SocketFlags
 
                     receivedFlags = messageHeader.Flags;
                     int sockAddrLen = messageHeader.SocketAddressLen;
-                    ipPacketInformation = GetIPPacketInformation(&messageHeader, isIPv4, isIPv6);
 
                     if (errno == Interop.Error.SUCCESS)
                     {
+                        ipPacketInformation = GetIPPacketInformation(&messageHeader, isIPv4, isIPv6);
                         socketAddressLen = sockAddrLen;
                         return checked((int)received);
                     }
                     else
                     {
+                        ipPacketInformation = default(IPPacketInformation);
```

Comment:
Not for this PR, but it'd be nice later to fix the structure of this code so that it matched the above code, or vice versa.  It's strange that the two pieces of code are identical in purpose but not in structure.